src/coreclr/tools/Common/TypeSystem/Common/TargetArchitecture.cs

```diff
@@ -14,6 +14,7 @@ public enum TargetArchitecture
         Unknown,
         ARM,
         ARM64,
+        ARMEL,
```

Comment:
What is this used for?

This would mean that all places the check for `TargetArchitecture.ARM` have to check for `TargetArchitecture.ARMEL` too?

Reply:
It is used for (easier) strongly typed binding. In Program.ctor, it overrides `ARMEL` -> `ARM` since the rest of the implementation expects it with a flag to indicate if `ARMEL` ABI is to be used.

Alternatively, I can use the old approach to capture the flag during parsing (it is already using custom step in call to `Option<T>` ctor for --targetarch).

Reply:
I think that would be better - since we do not expect this value to be valid, except in command line parsing.

Reply:
FWIW https://github.com/dotnet/runtime/pull/43706#discussion_r510754387 is the previous conversation on why we don't have TargetArchitecture.Armel.